src/core/tests/type_prop/rms_norm.cpp

```diff
@@ -0,0 +1,285 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/rms_norm.hpp"
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_assertions.hpp"
+#include "common_test_utils/type_prop.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/subtract.hpp"
+
+namespace ov {
+namespace test {
+
+using ov::op::v0::Constant;
+using ov::op::v0::Parameter;
+using testing::HasSubstr;
+
+class TypePropRMSNormTest : public TypePropOpTest<op::v14::RMSNorm> {};
+
+TEST_F(TypePropRMSNormTest, default_ctor) {
+    const auto op = make_op();
+    const auto data = std::make_shared<Parameter>(element::f16, PartialShape{2, 3, 8, 6});
+    const auto axes = std::make_shared<Parameter>(element::i64, PartialShape{1});
+    const auto scale = std::make_shared<Parameter>(element::f16, PartialShape{});
+
+    op->set_arguments(ov::OutputVector{data, axes, scale});
+    op->validate_and_infer_types();
+
+    EXPECT_EQ(op->get_output_size(), 1);
+    EXPECT_EQ(op->get_input_size(), 3);
+    EXPECT_EQ(op->get_output_element_type(0), element::f16);
+    EXPECT_EQ(op->get_output_partial_shape(0), (PartialShape{2, 3, 8, 6}));
+}
+
+TEST_F(TypePropRMSNormTest, no_scale_no_compute_type) {
+    const auto data = std::make_shared<Parameter>(element::f32, PartialShape{2, 3, 8, 6});
+    const auto axes = std::make_shared<Parameter>(element::i32, PartialShape{1});
+    const auto eps = 1e-5;
+
+    const auto op = make_op(data, axes, eps);
```

Comment:
In few tests where `compute_type` is not explicit specified by user get it as attribute and check if is same as `data` type as mention in this parameter description.

What if data will be dynamic type and compute type will be not specified?
What if `compute_type` will be specified by user as dynamic type (error or same behavior as for undefiend) ?

Reply:
There is no need to introduce logic that overwrite the `compute_type` member with the actual data type.
It can unnecessarily complicate RMSNorm op class.

The `undefined` is just a sign to not put additional Convert before and after reduce accumulation, what leads to usage of the same data type for computation. I would say the same behavior can be applied for `dynamic` element type.
For data Input with `dynamic` element type, there will be corresponding Tensor with static type during inference, so it's not an issue.

Let's proceed with this core op part, further concerns can be addressed within development of the reference implementation, so it will be easier to discuss the logic of the operation itself.

Reply:
But could be check if it has default value.